.github/workflows/ci.yml

```diff
@@ -43,54 +41,69 @@ jobs:
     strategy:
       matrix: ${{ fromJSON(needs.generate-matrix.outputs.matrix) }}
     steps:
-      - name: Checkout base repo
-        uses: actions/checkout@v4
       - name: "Install tools"
+        if: ${{ matrix.os == 'ubuntu-latest' }}
         run: |
-            .github/workflows/install-tools.sh
+          sudo apt install upx-ucl
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
       - name: Set up Haskell
         id: setup-haskell
         uses: haskell-actions/setup@v2
         with:
           ghc-version: ${{ matrix.ghc }}
           cabal-version: 'latest'
+
       - name: Configure
         run: |
           ARCHITECTURE=$(uname -m)
           echo "ARCH=$ARCHITECTURE" >> $GITHUB_ENV
           echo ${{ env.ARCH }}
           cabal configure --enable-tests
+
       - name: Freeze
         run: cabal freeze --project-file=cabal.release.project
+
       - name: Cache
-        uses: actions/cache@v4.0.2
+        uses: actions/cache@v4
         with:
           path: ${{ steps.setup-haskell.outputs.cabal-store }}
-          key: ${{ runner.os }}-ghc-${{ matrix.ghc }}-${{ hashFiles('**/plan.json') }}
-          restore-keys: ${{ runner.os }}-ghc-${{ matrix.ghc }}-
+          key: ${{ runner.os }}-ghc-${{ steps.setup-haskell.outputs.ghc-version }}-${{ hashFiles('**/plan.json') }}
+          restore-keys: ${{ runner.os }}-ghc-${{ steps.setup-haskell.outputs.ghc-version }}-
```

Comment:
*TLDR;* It's a leftover and I can revert the change if you like.

Long version: As I wrote, at some point I wanted to know where the different jobs in the different pipelines were different.
So I extracted those into four files (more or less like `{ci,release}-{ubuntu,alpine}.yml`) and aimed for a minimal diff of those files. Since the Alpine jobs use `steps.setup-haskell...` and the Ubuntu jobs use `matrix.ghc...` I needed to choose, and I decided to go with the first one.
Since this change is not related to this PR I am happy to revert it if you like.

Reply:
I'd be happy to consider these changes in their own PR at a later time indeed. Thank you for being understanding. :bow:

Reply:
Well, thank you for your patience!